code/game/machinery/doors/firedoor.dm

```diff
@@ -454,6 +454,8 @@
 /obj/structure/firelock_frame/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
 	if((constructionStep == CONSTRUCTION_NOCIRCUIT) && (the_rcd.upgrade & RCD_UPGRADE_SIMPLE_CIRCUITS))
 		return list("mode" = RCD_UPGRADE_SIMPLE_CIRCUITS, "delay" = 20, "cost" = 1)
+	else if(the_rcd.mode == RCD_DECONSTRUCT)

```

Comment:
i'd prefer this if check be first tbh, otherwise you'd always have to build up the frame first before you can deconstruct it if you meet the upgrade criteria